notifications/src/main/java/org/openedx/notifications/presentation/settings/NotificationsSettingsViewModel.kt

```diff
@@ -23,39 +32,72 @@ class NotificationsSettingsViewModel(
     val notificationsConfiguration: StateFlow<NotificationsConfiguration>
         get() = _notificationsConfiguration
 
+    private val _showPermissionRequestDialog = MutableStateFlow(false)
+    val showPermissionRequestDialog: StateFlow<Boolean>
+        get() = _showPermissionRequestDialog
+
+    private val _uiMessage = MutableSharedFlow<UIMessage>()
+    val uiMessage = _uiMessage.asSharedFlow()
+
     init {
-        viewModelScope.launch {
+        if (checkPushNotificationPermission()) {
             fetchAndUpdateNotificationsSettings()
+        } else {
+            updateDiscussionPreference(updatedValue = false)
+            _showPermissionRequestDialog.update { true }
         }
     }
 
     fun setDiscussionNotificationPreference(value: Boolean) {
-        viewModelScope.launch {
-            try {
-                val response = interactor.updateNotificationsConfiguration(value)
-                updateDiscussionPreference(updatedValue = response.updatedValue)
+        if (checkPushNotificationPermission()) {
```

Comment:
IMO, the app should display the backend notification switch state even if the user has not granted notification permissions. Additionally, an info dialog should appear when the user attempts to interact with the switch while notification permission is not granted.

Reply:
As pr AC, the switch will stay off if any of these 3 value is false.
- App notification permission
- data > discussion > enabled
- data > discussion > notification_types > core > push